cmd/derper/derper.go

```diff
@@ -177,14 +189,42 @@ func main() {
 	s.SetVerifyClientURLFailOpen(*verifyFailOpen)
 	s.SetTCPWriteTimeout(*tcpWriteTimeout)
 
-	if *meshPSKFile != "" {
-		b, err := os.ReadFile(*meshPSKFile)
+	if *dev {
+		k := setec.StaticSecret(os.Getenv("TAILSCALE_DERPER_MESH_KEY"))
+		if k.GetString() != "" {
+			key, err := checkMeshKey(k.GetString())
+			if err != nil {
+				log.Fatalf("setec static failed to get key: %v", err)
+			}
+			s.SetMeshKey(key)
+		} else {
+			log.Println("No mesh key specified for dev")
+		}
+	} else if *secretsURL != "" {
+		st, err := setec.NewStore(ctx,
+			setec.StoreConfig{
+				Client: setec.Client{Server: *secretsURL},
+				Secrets: []string{
+					*meshKeySecret,
+				},
+			})
 		if err != nil {
-			log.Fatal(err)
+			log.Fatalf("NewStore: %v", err)
+		}
+		k := st.Secret(*meshKeySecret)
+		key, err := checkMeshKey(k.GetString())
```

Comment:
```suggestion
		key, err := checkMeshKey(st.Secret(*meshKeySecret).GetString())
```

Though note you might also want to expand the name with a prefix here, depending on your thoughts about the above.